### PR TITLE
Set mypy strict_optional = True

### LINKED ISSUE #342: Set mypy strict_optional = True
### 🚀 Feature Request

Mypy should check optional values in strict way.

### Motivation

This way we can control type of the variables in more visible way.

### Proposal

Set strict_optional = True in `mypy.ini`
Change all the necessary code.

### Test cases

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [x] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by alex-hse-repository**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Closing issues

closes #342

### CODE REVIEW

File: etna/pipeline/pipeline.py [hunk 1]
```diff
@@ -283,6 +289,8 @@ def _get_backtest_metrics(self, aggregate_metrics: bool = False) -> pd.DataFrame
 
     def _get_fold_info(self) -> pd.DataFrame:
         """Get information about folds."""
+        if self._folds is None:
+            raise ValueError("Something went wrong during backtest initialisation!")
```
Comment (martins0n) [inline etna/pipeline/pipeline.py:293]: `initialisation` typo

**Review by martins0n — CHANGES_REQUESTED** [review]
👍 Some typos left

File: etna/pipeline/pipeline.py (hunk 1 above)
Comment (alex-hse-repository) [reply etna/pipeline/pipeline.py]: Fixed :see_no_evil:

**Review by martins0n — APPROVED** [review]
👍

**Pull request merged**